temporian/core/operators/lag.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Lag operator."""
+
+from temporian.core import operator_lib
+from temporian.core.data.event import Event
+from temporian.core.data.feature import Feature
+from temporian.core.operators.base import Operator
+from temporian.proto import core_pb2 as pb
+from temporian.core.data.duration import Duration
+
+
+class LagOperator(Operator):
+    """Lag operator."""
+
+    def __init__(
+        self,
+        event: Event,
+        duration: Duration,
+    ):
+        super().__init__()
+
+        # inputs
+        self.add_input("event", event)
+
+        self.add_attribute("duration", duration)
+
+        # outputs
+        output_features = [  # pylint: disable=g-complex-comprehension
+            Feature(
+                name=f"lag_{f.name()}",
+                dtype=f.dtype(),
+                sampling=event.sampling,
```

Comment:
To think about: should core `Feature`s keep having a reference to a core `Sampling`? We removed that in their NumPy counterparts. What are we gaining from core features having a core sampling, since they are attached to a core event that already has one? cc: @ianspektor @achoum